### PR TITLE
init commit

### PR DESCRIPTION
**Pull request by entertainment-veks**

### CODE REVIEW

File: src/main/java/hibernate/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -45,4 +45,14 @@ public Movie add(Movie movie) {
...
     }
+
+    @Override
+    public void delete(Movie movie) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Transaction transaction = session.beginTransaction();
+            session.remove(movie);
+            transaction.commit();
+        }
+        System.out.println(String.format("Movie with id %s should be removed", movie.getId()));
```
Comment (NikolayFatkullin) [inline src/main/java/hibernate/dao/impl/MovieDaoImpl.java:56]: we don't need this method in this task, but anyway you need to add a catch block

File: src/main/java/hibernate/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public List<MovieSession> getAllInTime(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> query = session.createQuery("from movie_session"
```
Comment (NikolayFatkullin) [inline src/main/java/hibernate/dao/impl/MovieSessionDaoImpl.java:19]: I think it better to use the class name, not the table name here, but I'm not sure

**Review by NikolayFatkullin — CHANGES_REQUESTED** [review]

**Review by NikolayFatkullin — APPROVED** [review]

File: src/main/java/hibernate/dao/MovieDao.java [hunk 3]
```diff
@@ -7,4 +7,6 @@
     Movie add(Movie movie);
 
     List<Movie> getAll();
+
+    void delete(Movie movie);
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/dao/MovieDao.java:11]: let's remove this

File: src/main/java/hibernate/dao/MovieSessionDao.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package hibernate.dao;
+
+import hibernate.model.MovieSession;
+import java.time.LocalDate;
+import java.util.List;
+
+public interface MovieSessionDao {
+    List<MovieSession> getAllInTime(Long movieId, LocalDate date);
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/dao/MovieSessionDao.java:8]:
```suggestion
    List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
```

File: src/main/java/hibernate/dao/impl/CinemaHallDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaQuery<CinemaHall> criteriaQuery = session.getCriteriaBuilder()
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/dao/impl/CinemaHallDaoImpl.java:40]: let's use better use HQL in all DaoImpl classes

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/hibernate/dao/impl/CinemaHallDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> query = session.createQuery("from cinema_hall", CinemaHall.class);
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all cinema halls", e);
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/dao/impl/CinemaHallDaoImpl.java:43]: let's use CustomDaoException

File: src/main/java/hibernate/dao/impl/MovieSessionDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+                    + " WHERE movie.id = :movieId"
+                    + " AND DATE_FORMAT(showTime, '%Y-%m-%d') = :date",
+                    MovieSession.class);
+            query.setParameter("movieId", movieId);
+            query.setParameter("date", date.toString());
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find movie session by params: "
```
Comment (KseniiaMakarova) [inline src/main/java/hibernate/dao/impl/MovieSessionDaoImpl.java:27]: same

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**